### ISSUE TITLE
CLH CI regression for kata 2.0 in the tests repo (2.0-dev branch)

### ISSUE THREAD

**Issue by likebreath** [kata-containers/tests#2863]
We observe regressions on the CLH CI for our tests repo: http://jenkins.katacontainers.io/job/kata-containers-2.0-tests-ubuntu-clh-virtiofs-PR/. The main reason is this CI is not tested regularly when merging PR. The last success run was on August 26 [here](http://jenkins.katacontainers.io/job/kata-containers-2.0-tests-ubuntu-clh-virtiofs-PR/14/), and has been tested only 6 times (all failing) since that then. The most recent failing from run [22](http://jenkins.katacontainers.io/job/kata-containers-2.0-tests-ubuntu-clh-virtiofs-PR/22/), which is likely to be a CI setup issue as shown below:

```
02:53:34 INFO: Build agent
02:53:34 + pushd /tmp/jenkins/workspace/kata-containers-2.0-tests-ubuntu-clh-virtiofs-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/rootfs-builder/../../../src/agent/
02:53:34 /tmp/jenkins/workspace/kata-containers-2.0-tests-ubuntu-clh-virtiofs-PR/go/src/github.com/kata-containers/kata-containers/src/agent /tmp/jenkins/workspace/kata-containers-2.0-tests-ubuntu-clh-virtiofs-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder
02:53:34 + '[' -n '' ']'
02:53:34 + info 'checkout failed!'
02:53:34 + local 'msg=checkout failed!'
02:53:34 + echo 'INFO: checkout failed!'
02:53:34 INFO: checkout failed!
02:53:34 + make clean
02:53:34 make[1]: Entering directory '/tmp/jenkins/workspace/kata-containers-2.0-tests-ubuntu-clh-virtiofs-PR/go/src/github.com/kata-containers/kata-containers/src/agent'
02:53:34 make[1]: cargo: Command not found
02:53:34 Makefile:122: recipe for target 'clean' failed
02:53:34 make[1]: *** [clean] Error 127
02:53:34 make[1]: Leaving directory '/tmp/jenkins/workspace/kata-containers-2.0-tests-ubuntu-clh-virtiofs-PR/go/src/github.com/kata-containers/kata-containers/src/agent'
02:53:34 ++ error_handler
02:53:34 ++ '[' 2 -eq 0 ']'
02:53:34 ++ '[' -n '' ']'
02:53:34 Makefile:87: recipe for target '/tmp/jenkins/workspace/kata-containers-2.0-tests-ubuntu-clh-virtiofs-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/.ubuntu_rootfs.done' failed
02:53:34 make: *** [/tmp/jenkins/workspace/kata-containers-2.0-tests-ubuntu-clh-virtiofs-PR/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/.ubuntu_rootfs.done] Error 2
02:53:34 Failed at 22: "${cidir}/install_kata_image.sh" "${tag}"
02:53:34 Build step 'Execute shell' marked build as failure
02:53:42 Performing Post build task...
```

**Comment by likebreath** [comment]
For future reference, here is the summary of the fix from PR #2862: 

This PR fixes the regression on the CLH CI in the test repo for kata 2.0 ( http://jenkins.katacontainers.io/job/kata-containers-2.0-tests-ubuntu-clh-virtiofs-PR/). There are three issues being fixed:

1. In many of our CI scripts, we assume the 'KATA_DEV_MODE' environment variable to be not set at all when we are not in the 'kata dev mode'. Setting its value explicitly to 'false' would fail the check '[-n $KATA_DEV_MODE]' and make the script believe we ARE in the 'kata dev mode', which would skip some installation. For example, 'install_rust.sh' is totally skipped in CLH's CI, which caused rust-agent compiling failure and the CI failure. (commit 26e49ad).

2. Failure on the new k8s test pod OOM. As this is a new k8s tests, it is advisable to debug on this in a separate PR, after bring CLH CI back online. (commit d42d7ae)

3. Build failure for cloud-hypervisor. (commit kata-containers/kata-containers@7c1c1ff)

**Comment by likebreath** [comment]
This issue is now being fixed, as two PRs are landed: #2862 and https://github.com/kata-containers/kata-containers/pull/762.
